core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -344,7 +344,7 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 			return nil, nil, err1
 		}
 
-		if chainID == destChainID {
+		if chainID == destChainID && destChainFamily == relayChainFamily {
```

Comment:
Whats the difference between a chainID and a chain selector?

Is it true that this extra check is to ensure we don't mis-match in case the same chainID is used in different chain families?

Reply:
Yes, chain selector randomly generated uint64 and used by CCIP, and it is unique across all chains/families.

The chain id is specific to a chain family, so it doesn't provide this guarantee. For example `chainid`: `1` can either be`ethereum-mainnet` or `aptos-mainnet`

Reply:
Thanks, I was mixing up chainID and chainSelector. They're both differen't ways to ID a chain so it's hard to reason about, some kind of (chainID, chainFamily) tuple type and/or converting to one or the other rather than passing them around is needed. Having an ID with type string doesn't help.

I believe `getChainReaderConfig` has this same bug.